opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchStorageEngine.java

```diff
@@ -9,27 +9,48 @@
 import static org.opensearch.sql.utils.SystemIndexUtils.isSystemIndex;
 
 import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.CatalogSchemaName;
+import org.opensearch.sql.analysis.model.CatalogSchemaIdentifierName;
+import org.opensearch.sql.analysis.model.SchemaName;
 import org.opensearch.sql.common.setting.Settings;
+import org.opensearch.sql.exception.SemanticCheckException;
 import org.opensearch.sql.opensearch.client.OpenSearchClient;
 import org.opensearch.sql.opensearch.storage.system.OpenSearchSystemIndex;
 import org.opensearch.sql.storage.StorageEngine;
 import org.opensearch.sql.storage.Table;
+import org.opensearch.sql.utils.SystemIndexUtils;
 
-/** OpenSearch storage engine implementation. */
+/**
+ * OpenSearch storage engine implementation.
+ */
 @RequiredArgsConstructor
 public class OpenSearchStorageEngine implements StorageEngine {
 
-  /** OpenSearch client connection. */
+  /**
+   * OpenSearch client connection.
+   */
   private final OpenSearchClient client;
 
   private final Settings settings;
 
   @Override
-  public Table getTable(String name) {
-    if (isSystemIndex(name)) {
-      return new OpenSearchSystemIndex(client, name);
+  public Table getTable(CatalogSchemaName catalogSchemaName, String tableName) {
+    if (isSystemIndex(tableName)) {
+      return new OpenSearchSystemIndex(client, tableName);
+    } else if (SchemaName.INFORMATION_SCHEMA_NAME.equals(catalogSchemaName.getSchemaName())) {
```

Comment:
Relook into this.

Reply:
why this change is needed? Does the SchemaName.INFORMATION_SCHEMA_NAME.equals code path used by any execution logic?

Reply:
`source = prometheus.information_schema.tables`  is the command to list all the prometheus metrics.

Since it is a minor change on opensearch side to support information_schema, I have made these changes. We are just reusing the execution of SHOW TABLES;

if we want to use the same for default storage engine, we can write @opensearch.information_schema.tables.

Reply:
which means we will support `source = .opensearch.information_schema.tables`, if so, make sense. add it to the docs?

Reply:
Yes, raising docs PR in a while. 

I have changed the default opensearch cluster name to @opensearch since we are using . to separate identifiers causing confusion and complex code logic.

Also I will propose new design for information schema for next release. Will be  implementing information_schema as a separate internal connector similar to opensearch, prometheus, s3.

Will include more apis to StorageEngine interface to list tables and list columns.